### ISSUE TITLE
The "previous_release" available only after deploy:release task.

### ISSUE THREAD

**Issue by coreyworrell** [deployphp/deployer#3380]
- Deployer version: 7.0.2
- Deployment OS: Alpine

The first deployment always fails because of this error.

```php
task('deploy:update_code', function (): void {
    if (has('previous_release')) {
        run('cp -aT {{previous_release}}/ {{release_path}}/ || true');
    }

    // ...rest of task
})

task('deploy', [
    'deploy:setup',
    'deploy:lock',
    'deploy:release',
    'deploy:update_code',
    'deploy:shared',
    'deploy:writable',
    'deploy:clear_paths',
    'deploy:symlink',
    'deploy:unlock',
    'deploy:cleanup',
    'deploy:success',
]);
```

`deploy:release` is running before `deploy:update_code` which makes the call to `has('previous_release')` so I'm not sure why this happens. Is there something missing here? Running the deploy again after works.

**Comment by kfrohwein** [comment]
Deployer version: 7.0.2
Deployment OS: Ubuntu 20.04.5 LTS

Same for me. I have deleted the releaes and .dep folder because I am testing for PROD on my Stage. The log shows 1 release. Running it again as stated works. I guess the config is just not set on the first run?

**Comment by Nickolaus** [comment]
` has('previous_release')` seems to become `true` after `task deploy:release` has been performed

My workaround is presetting it upon init:
`set('is_first_release', has('previous_release'));`

then you can do something like this:
`if (!get('is_first_release'))`

**Comment by coreyworrell** [comment]
@Nickolaus the problem is that just calling the function `has('previous_release')` will throw an error on first deploy, even after `deploy:release` has been run. Are you saying that calling `has()` outside of a task, in initial configuration does not throw an error on first deploy?

**Comment by Nickolaus** [comment]
@coreyworrell yes in my case if I write the value initially into a variable, I can preserve the inital result of `previous_release` which at this time actually is `true` and then later on I am using the value of the variable to check

**Comment by coreyworrell** [comment]
@Nickolaus that sounds like it could work as a workaround, thank you. But it still seems like a bug that my example in original post does not work. Hopefully @antonmedv can chime in.

**Comment by antonmedv** [comment]
Will try to take a look on this weekend.

**Comment by adriaanzon** [comment]
I am also getting this error when deploying new servers. In the sentry recipe there is a [`has('previous_release')`](https://github.com/deployphp/deployer/blob/1f5cfb25c247506b8d2a12484c87f9b8b3fe3f9b/contrib/sentry.php#L194) check, but this check will always return true since 'previous_release' is always set (this was added in https://github.com/deployphp/deployer/commit/ed626d4b1936289b730272cc872c6d08f3880e0c).

**Comment by coreyworrell** [comment]
Yeah I don't understand why it needs to throw an exception if there's no previous release. It should just be empty.

**Comment by antonmedv** [comment]
Fixed in 7.1
